sense-android-library/src/nl/sense_os/service/EncryptionHelper.java

```diff
@@ -0,0 +1,121 @@
+package nl.sense_os.service;
+
+import java.security.GeneralSecurityException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.telephony.TelephonyManager;
+import android.util.Base64;
+import android.util.Log;
+
+public class EncryptionHelper {
+
+	public static class EncryptionHelperException extends RuntimeException {
+
+		public EncryptionHelperException(Throwable e) {
+			super(e);
+		}
+
+	}
+
+	private static final String TAG = "EncryptionHelper";
+
+	private static final String TRANSFORMATION = "AES/CBC/PKCS5Padding";
+	private static final String SECRET_KEY_HASH_TRANSFORMATION = "SHA-256";
+	private static final String CHARSET = "UTF-8";
+	private static final String KEY_SALT = "3XnMxOy3E&jsdSSHWM941D89yK!RlRVH";
```

Comment:
@orhan89 We should also a preference for the salt, and if non is given by the application take this default one. Because the sense-android-library is public having the salt the same for every application does not make sense.